### PR TITLE
add equiv check to filebeat as shipper test

### PR DESCRIPTION
**Pull request by leehinman**
## Proposed commit message

add document equivalency check to filebeat as shipper process integration test

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
~~- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

### Run go integration tests

```
cd x-pack/filebeat
mage goUnitTest
```

### Just run this test

```
cd x-pack/filebeat
mage docker:composeUp
go test ./tests/integration -count 1 -tags integration -run TestShipperInputOutput
```

## Related issues

- Closes elastic/elastic-agent-shipper#290

### CODE REVIEW

File: x-pack/filebeat/tests/integration/shipper_test.go [hunk 1]
```diff
@@ -185,6 +212,61 @@ processors:
 		require.NoError(t, err, "error doing search request: %s", err)
 		return res.Hits.Total.Value == 2
 	}, 30*time.Second, 250*time.Millisecond, "never found 2 documents")
-	// ToDo add comparison of docs to make sure they are the same
-	// for example right now input.type is being overwritten with shipper
+
+	res, err := es.Search().
```
Comment (belimawr) [inline x-pack/filebeat/tests/integration/shipper_test.go:216]: This error here is not checked, which result in the test failing without a clear reason.

File: x-pack/filebeat/tests/integration/shipper_test.go [hunk 2]
```diff
@@ -185,6 +212,61 @@ processors:
...
+	f1 := mapstr.M(d1).Flatten()
+	f2 := mapstr.M(d2).Flatten()
+
+	for _, key := range fieldsToDrop {
+		_ = f1.Delete(key)
+	}
+
+	for _, key := range fieldsToDrop {
+		_ = f2.Delete(key)
+	}
```
Comment (belimawr) [inline x-pack/filebeat/tests/integration/shipper_test.go:263]: It's just a matter of style, but that could be a single for loop:
```suggestion
	for _, key := range fieldsToDrop {
		_ = f1.Delete(key)
		_ = f2.Delete(key)
	}
```

**Review by belimawr — CHANGES_REQUESTED** [review]

File: x-pack/filebeat/tests/integration/shipper_test.go (hunk 1 above)
Comment (leehinman) [reply x-pack/filebeat/tests/integration/shipper_test.go:216]: thanks, fixed

File: x-pack/filebeat/tests/integration/shipper_test.go (hunk 2 above)
Comment (leehinman) [reply x-pack/filebeat/tests/integration/shipper_test.go]: thanks, reduced to single loop

**Comment by leehinman** [comment]
@cmacknz or @pierrehilbert can you review?  I think the change to go.mod triggered a review by Beats Tech Leads

**Review by cmacknz — APPROVED** [review]

**Pull request merged**